esp/src/src-react/comms/fileSpray.ts

```diff
@@ -0,0 +1,342 @@
+import { FileSprayService, TopologyService } from "@hpcc-js/comms";
+import { scopedLogger } from "@hpcc-js/util";
+import nlsHPCC from "src/nlsHPCC";
+import { Thenable } from "src/store/Deferred";
+import { Paged } from "src/store/Paged";
+import { BaseStore } from "src/store/Store";
+
+const logger = scopedLogger("src-react/comms/fileSpray.ts");
+
+const fileSprayService = new FileSprayService({ baseUrl: "" });
+const topologyService = new TopologyService({ baseUrl: "" });
+
+/**
+ * Utility function to encode logical file paths
+ * Replicates the lfEncode function from legacy FileSpray.ts
+ */
+export const lfEncode = (path: string): string => {
+    let retVal = "";
+    for (let i = 0; i < path.length; ++i) {
+        switch (path[i]) {
+            case "/":
+            case "\\":
+                retVal += "::";
+                break;
+            case "A":
+            case "B":
+            case "C":
+            case "D":
+            case "E":
+            case "F":
+            case "G":
+            case "H":
+            case "I":
+            case "J":
+            case "K":
+            case "L":
+            case "M":
+            case "N":
+            case "O":
+            case "P":
+            case "Q":
+            case "R":
+            case "S":
+            case "T":
+            case "U":
+            case "V":
+            case "W":
+            case "X":
+            case "Y":
+            case "Z":
+                retVal += "^" + path[i];
+                break;
+            default:
+                retVal += path[i];
+        }
+    }
+    return retVal;
+};
+
+export interface DropZoneMachine {
+    Netaddress: string;
+    ConfigNetaddress: string;
+    Directory: string;
+    OS: number;
+}
+
+export interface DropZone {
+    Name: string;
+    Path: string;
+    Linux: string;
+    machine: DropZoneMachine;
+    TpMachines?: {
+        TpMachine: DropZoneMachine[];
+    };
+}
+
+export interface FileListItem {
+    calculatedID: string;
+    name: string;
+    isDir: boolean;
+    filesize?: number;
+    modifiedtime?: string;
+    NetAddress: string;
+    OS: number;
+    fullPath: string;
+    fullFolderPath: string;
+    DropZone: DropZone;
+    displayName: string;
+    type: string;
+    _isUserFile?: boolean;
+    getLogicalFile(): string;
+}
+
+export type LandingZonesStore = BaseStore<any, DropZone>;
```

Comment:
The "any" should be request interface for the service (again could be pulled from comms?)